src/main/java/minijava/ir/assembler/SimpleNodeAllocator.java

```diff
@@ -0,0 +1,171 @@
+package minijava.ir.assembler;
+
+import static minijava.ir.utils.FirmUtils.getMethodLdName;
+
+import firm.Graph;
+import firm.Mode;
+import firm.TargetValue;
+import firm.nodes.*;
+import java.util.*;
+import minijava.ir.assembler.instructions.Argument;
+import minijava.ir.assembler.instructions.ConstArgument;
+import minijava.ir.assembler.location.Location;
+import minijava.ir.assembler.location.Register;
+import minijava.ir.assembler.location.StackLocation;
+import minijava.ir.utils.MethodInformation;
+
+/**
+ * Simple stack based node allocator that tries to implement the scheme from the compiler lab
+ * slides.
+ *
+ * <p>Every value is stored on the stack in a new slot thereby using lot's of memory. The advantage
+ * of this inefficiency is the simplicity of the implementation.
+ */
+public class SimpleNodeAllocator implements NodeAllocator {
+
+  private static final int STACK_SLOT_SIZE = 8; // we're on an 64bit system
+  private Map<Node, Integer> assignedStackSlots;
+  private int currentSlotNumber = 1;
+  private Graph graph;
+  private MethodInformation info;
+
+  @Override
+  public void process(Graph graph) {
+    this.graph = graph;
+    this.info = new MethodInformation(graph);
+    this.assignedStackSlots = new HashMap<>();
+  }
```

Comment:
Ah, now the purpose of the interface method is more clear.

Do we really need this `process` method? This just seems to initialize stuff and hide the graph instance, which I'd rather see passed as an explicit argument to methods which need it. As I said, I really dislike instance state which leaks out of an interface.